### ISSUE TITLE
This + badges() in messages needs to be addressed before the final release.

### ISSUE THREAD

**Issue by museun** [museun/twitchchat#79]
https://github.com/museun/twitchchat/blob/1b865fdc719087887a9183fb95e980afb1cf248c/src/twitch/mod.rs#L29-L31

**Comment by museun** [comment]
After playing around with `std::borrow::Cow` for a bit, I think switching from the generic `T: StringMarker` w/ `Conversion` helper trait, and sometimes using an elided lifetime would be more ergonomic.

```rust
struct Foo<T: StringMarker = String> {
    data: T
}
// could be
struct Foo<'t> {
   data: Cow<'t, str>
}
```

the `Conversion` trait could still be used to do:
```rust
// realloc the string onto the heap and then turn it back into a Cow
let data : Cow<'static, str> = foo.data.map(|s| s.deref().to_owned().into())
```
without so much typing.

Most of the code could elide the lifetimes `Foo<'_>` or request `Foo<'static>` but I haven't spent much time in figuring out how the monomorphization that the current scheme does will work with this. Having an enum (the Cow) will always be runtime dispatching..

**Comment by museun** [comment]
Implemented in #81
